test/Lazy.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+var FL = require('fantasy-land');
+
+
+//  Lazy :: (() -> a) -> Lazy a
+function Lazy(f) {
+  if (!(this instanceof Lazy)) return new Lazy(f);
+  this.run = f;
+}
+
+Lazy[FL.of] = function(a) {
+  return Lazy(function() { return a; });
+};
+
+Lazy.prototype['@@type'] = 'sanctuary-type-classes/Lazy';
+
+Lazy.prototype[FL.map] = function(f) {
+  return this[FL.ap](Lazy[FL.of](f));
+};
+
+Lazy.prototype[FL.ap] = function(other) {
+  var f = other.run;
+  var v = this.run;
+  return Lazy(function() { return f()(v()); });
```

Comment:
What do you think of this change?

```javascript
  var lazy = this;
  return Lazy(function() { return other.run()(lazy.run()); });
```

I like to avoid variables where possible. Perhaps you wanted to avoid accessing the `run` property multiple times, though?